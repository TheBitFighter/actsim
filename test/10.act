defproc dendrites (chan?(bool) tick; chan?(int) W, Vi; chan!(int) Vo)
{
    int wgt, in, sum;
    bool tk;
    chp
    {
        sum:=0;
        log("in dend"); 
       *[
            [ #W|#Vi -> W?wgt, Vi?in; log("Recv w v:", wgt," ",in); sum:=sum+(wgt*in)
            [] #tick -> tick?tk; log("got tick"); Vo!sum; log("Send S:",sum); sum:=0
            ]
         ]
    }
}


defproc drive(chan!(int) w, v; chan!(bool) tick; chan?(int) s)
{
    bool tk;
    int i, v_val, w_val, sum;
    chp
    {
        tk:=false;
        i:=0;
        v_val:=15;
        w_val:=2;
     *[ [i < 5 -> tick!tk; s?sum; log("Send w v:",w_val," ",i); i:=i+1] ]
    }
}


defproc test()
{
    dendrites dn;
    drive dv(dn.W, dn.Vi, dn.tick, dn.Vo);
}
