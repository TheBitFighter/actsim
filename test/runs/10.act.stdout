[      20 t#:0] <dn> in dend
[      60 t#:0] <dn> got tick
[      70 t#:0] <dn> Send S:0
[      70 t#:0] <dv> Send w v:2 0
[      90 t#:0] <dn> got tick
[     100 t#:0] <dn> Send S:0
[     100 t#:0] <dv> Send w v:2 1
[     120 t#:0] <dn> got tick
[     130 t#:0] <dn> Send S:0
[     130 t#:0] <dv> Send w v:2 2
[     150 t#:0] <dn> got tick
[     160 t#:0] <dn> Send S:0
[     160 t#:0] <dv> Send w v:2 3
[     180 t#:0] <dn> got tick
[     190 t#:0] <dn> Send S:0
[     190 t#:0] <dv> Send w v:2 4
